test/assertion/assert_unary_iteration/test_assert_unary_iteration.py

```diff
@@ -0,0 +1,261 @@
+    Ran 9 tests in 0.021s
+
+    OK
+    $
+    """
+
+    def test_return_none(self,):
+
+        qc = QuantestPyCircuit(5)
+        qc.add_gate({"name": "z", "control_qubit": [0], "target_qubit": [2],
+                    "control_value": [1]})
+        qc.add_gate({"name": "y", "control_qubit": [0, 1], "target_qubit": [3],
+                    "control_value": [1, 1]})
+
+        self.assertIsNone(
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                system_reg=[2, 3, 4],
+                input_to_output={
+                    "00": "000",
+                    "01": "000",
+                    "10": "100",
+                    "11": "110"
+                }
+            )
+        )
+
+    def test_return_assert_error(self,):
+
+        qc = QuantestPyCircuit(5)
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+        qc.add_gate({"name": "z", "control_qubit": [2], "target_qubit": [3],
+                    "control_value": [1]})
+        qc.add_gate({"name": "y", "control_qubit": [2], "target_qubit": [4],
+                    "control_value": [1]})
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+
+        expected_error_msg = "In bitstring: 10\n" \
+            + "Out bitstring expect: 10\n" \
+            + "Out bitstring actual: 00"
+        with self.assertRaises(QuantestPyAssertionError) as cm:
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                system_reg=[3, 4],
+                input_to_output={
+                    "00": "00",
+                    "01": "00",
+                    "10": "10",
+                    "11": "11"
+                }
+            )
+        self.assertEqual(cm.exception.args[0], expected_error_msg)
+
+    def test_return_ancilla_error(self,):
+
+        qc = QuantestPyCircuit(5)
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+        qc.add_gate({"name": "z", "control_qubit": [2], "target_qubit": [3],
+                    "control_value": [1]})
+        qc.add_gate({"name": "y", "control_qubit": [2], "target_qubit": [4],
+                    "control_value": [1]})
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 0]})
+
+        expected_error_msg = "In bitstring: 10\n" \
+            + "Qubits [2] in ancilla reg are not back to 0 by uncomputation."
+        with self.assertRaises(QuantestPyAssertionError) as cm:
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                ancilla_reg=[2],
+                system_reg=[3, 4],
+                input_to_output={
```

Comment:
`system_reg` should be before `ancilla_reg`.

Reply:
👉 https://github.com/QuantestPy/quantestpy/pull/187/commits/af02c3fe010721ff967a38662bbb75cceeefdb76